### ISSUE TITLE
bazaar: error build

### ISSUE THREAD

**Issue by Izorkin** [NixOS/nixpkgs#44260]
## Issue description
error - 
```
trying https://bazaar.launchpad.net/~brz/brz/trunk/revision/6754
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  7810    0  7810    0     0   6929      0 --:--:--  0:00:01 --:--:--  6929
error: Normalized patch '/tmp/nix-build-CVE-2017-14176.patch.drv-0/0mdqa9w1p6cmli6976v4wi0sw9r4p5prkj7lzfd1877wk11c9c73' is empty (while the fetched file was not)!
```
full output error - https://pastebin.com/pFxEfAkx

### Steps to reproduce
Clear cache
nix build -f ~/nixpkgs-unstable bazzar

## Technical details
 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.54, NixOS, 18.09pre145679.dae9cf6106d (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.1pre6148_a4aac7f`
 - channels(root): `"nixos-18.09pre147750.94cbb8e3cd6"`
 - channels(lf): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by srhb** [comment]
```
    (fetchpatch {
      url = "https://bazaar.launchpad.net/~brz/brz/trunk/revision/6754";
      sha256 = "0mdqa9w1p6cmli6976v4wi0sw9r4p5prkj7lzfd1877wk11c9c73";
      name = "CVE-2017-14176.patch";
    })
```

This URL seems like it should be pointing to a diff, not a revision.

**Comment by xeji** [comment]
fixed in #44298
